internal/app/upstream/client.go

```diff
@@ -25,28 +45,26 @@ type Client interface {
 	// If the timeouts are exhausted, receive fails or a irrecoverable error occurs, the error is sent back to the caller.
 	// It is the caller's responsibility to send a new request from the last known DiscoveryRequest.
 	// Cancellation of the context cleans up all outstanding streams and releases all resources.
-	OpenStream(context.Context, *v2.DiscoveryRequest, string) chan *Response
+	OpenStream(*v2.DiscoveryRequest) (<-chan *v2.DiscoveryResponse, chan bool, error)
 }
 
 type client struct {
```

Comment:
Would also add the logger to this struct and log on the error received upstream since we're no longer propagating downstream.